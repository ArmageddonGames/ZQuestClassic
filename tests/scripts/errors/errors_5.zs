#option NO_ERROR_HALT on
#option WARN_DEPRECATED error

global script Active
{
	void run()
	{
		//Read-only vars should warn
		Game->CurScreen = 2;
		Game->CurDMap = 5;
		Debug->NULL = 0;
		//Deprecated things error due to #option
		Untype(5);
		TraceNPC(Screen->LoadNPC(0));
		Trace(Screen->ScriptEntry);
	}
}

