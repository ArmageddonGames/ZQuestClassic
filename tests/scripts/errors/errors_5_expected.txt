Compiling 'errors_5.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
ZQ_BUFFER Line 9 @ Columns 3-18 - Warning C105: 'Game->CurScreen' is read-only, and cannot be written to!
ZQ_BUFFER Line 10 @ Columns 3-16 - Warning C105: 'Game->CurDMap' is read-only, and cannot be written to!
ZQ_BUFFER Line 11 @ Columns 3-14 - Warning C105: 'Debug->NULL' is read-only, and cannot be written to!
ZQ_BUFFER Line 13 @ Columns 3-12 - Error S095: Function 'Untype(untyped)' is deprecated, and cannot be used due to current compiler settings.
INFO: Use casting '<untyped>(var)' instead!
ZQ_BUFFER Line 14 @ Columns 3-31 - Error S095: Function 'TraceNPC(NPC)' is deprecated, and cannot be used due to current compiler settings.
INFO: Use 'Trace()' instead!
ZQ_BUFFER Line 15 @ Columns 9-28 - Error S095: Variable 'Screen->ScriptEntry' is deprecated, and cannot be used due to current compiler settings.
Failure!
Compile finished with exit code '95' (compiled with errors)
